cmd/kueuectl/app/list/list_workload.go

```diff
@@ -160,6 +173,37 @@ func (o *WorkloadOptions) Complete(clientGetter util.ClientGetter, cmd *cobra.Co
 		return err
 	}
 
+	if o.UserSpecifiedForObject != "" {
+		mapper, err := clientGetter.ToRESTMapper()
+		if err != nil {
+			return err
+		}
+		var found bool
+		o.forGVK, o.forName, found, err = decodeResourceTypeName(mapper, o.UserSpecifiedForObject)
+		if err != nil {
+			return err
+		}
+		if !found {
+			return errors.New("--for must be in resource/name form")
+		}
+
+		r := clientGetter.NewBuilder().
+			Unstructured().
+			NamespaceParam(o.Namespace).
+			DefaultNamespace().
```

Comment:
do we need this line? should be before `NamespaceParam`?

Reply:
I think we need it. As I understand `DefaultNamespace()` assures all the objects retrieved have a namespace valud and uses `NamespaceParam` when it's empty. 

Here are a couple of examples where `NamespaceParam` and `DefaultNamespace` are used together:
- https://github.com/kubernetes/kubectl/blob/f74c97d4abe076617b583a6830c7ef1d7e961a0a/pkg/cmd/get/get.go#L470
- https://github.com/kubernetes/kubectl/blob/f74c97d4abe076617b583a6830c7ef1d7e961a0a/pkg/cmd/describe/describe.go#L175